src/sw/sw.ts

```diff
@@ -18,4 +18,23 @@ const pushEventHandler = {
         ));
   }
 };
+
 self.addEventListener('push', pushEventHandler);
+
+type CustomNotification = {
+  data: NotificationURLData|void;
+};
+
+const clickEventHandler = {
+  handleEvent: (event: NotificationEvent) => {
+    event.notification.close();
+
+    // tslint:disable-next-line:no-any
+    const notification = ((event.notification as any) as CustomNotification);
+    if (notification.data && notification.data.url) {
+      event.waitUntil(self.clients.openWindow(notification.data.url));
```

Comment:
What does this do?

Reply:
I've broken this line up and commented it. Essentially it opens a window and ensures the SW isn't killed until the window has opened.